src/pseudopeople/interface.py

```diff
@@ -97,6 +100,35 @@ def _generate_dataset(
     return noised_dataset
 
 
+def validate_source_compatibility(source: Path):
+    # TODO: Clean this up w/ metadata
+    changelog = source / "CHANGELOG.rst"
+    if changelog.exists():
+        with open(changelog, "r") as file:
+            first_line = file.readline()
+        version = first_line.split("**")[1].split("-")[0].strip()
+        # TODO: Implement a more robust check on version numbers
+        # TODO: log pseudopeople version where appropriate
+        if version > "1.4.2":
+            raise DataSourceError(
+                f"The provided simulated population data is incompatible with this version of pseudopeople ({psp_version}).\n"
+                "A newer version of simulated population data has been provided.\n"
+                "Please upgrade the pseudopeople package."
+            )
+        if version < "1.4.2":
```

Comment:
There are enough of these compares on version it should get a fix IMHO.

Reply:
Fixed. Left the TODO in to clean it up w/ a metadata file but this gets us through release